subtrack.MAUI/Services/SettingsService.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Microsoft.EntityFrameworkCore;
+using subtrack.DAL;
+using subtrack.DAL.Entities;
+using subtrack.MAUI.Services.Abstractions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace subtrack.MAUI.Services
+{
+    public class SettingsService : ISettingsService
+    {
+        private readonly SubtrackDbContext _context;
+        public async Task<T> GetByIdAsync<T>(string id) where T : SettingsBase
+        {
+            return await _context.Settings.OfType<T>().FirstOrDefaultAsync(x=>x.Id==id);
+        }
+
+        public async Task UpdateAsync<T>(T setting) where T : SettingsBase
+
+        {
+            _context.Update(setting);
```

Comment:
Does .Update call .SaveChanges?

Reply:
My bad, from what I understand it begins tracking the specified entity in modified state but no changes are made until SaveChangesAsync() is actually called. https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.dbcontext.update?view=efcore-7.0 https://learn.microsoft.com/en-us/ef/core/change-tracking/#how-to-track-entities